client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +160,35 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+export const deleteComment = ( { dispatch }, action ) => {
+	const { siteId, commentId } = action;
+	dispatch(
+		http( {
+			method: 'POST',
+			apiVersion: '1.1',
+			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			onSuccess: {
+				...action,
+				type: COMMENTS_DELETE_SUCCESS,
+			},
+			onFailure: {
+				...action,
+				type: COMMENTS_DELETE_FAILURE,
+			},
+		} )
+	);
+};
+
+export const removeCommentFromState = ( { dispatch }, action ) => {
+	const { siteId, postId, commentId } = action;
+	dispatch( removeComment( siteId, postId, commentId ) );
```

Comment:
> Is there a reducer listening to that action type?

Currently (in master) `COMMENTS_DELETE` in only defined in the action types, but never used anywhere.
I haven't included it in a reducer because there already is `COMMENTS_REMOVE` that removes comments from the state.

> How do you know they aren't firing?

I've added a log at the beginning of each functions and they work as expected, so I know that the functions are called correctly.

Then, though, when the delete is successful:
- No `COMMENTS_REMOVE` show up in the dev tools.
- The comment list is unchanged, so nothing has been removed.
And when it fails:
- No `CREATE_NOTICE` show up.
- No error notice pops up with the failure message.